.github/workflows/periodic_ci.yml

```diff
@@ -1,58 +1,41 @@
-name: Periodic
+name: Periodic CI
 
 on:
   schedule:
     - cron: '0 */6 * * *'
+  push:
+    paths:
+      - '**/periodic_ci.yml'
+  pull_request:
+    paths:
+      - '**/periodic_ci.yml'
 
-jobs:
-  network-tests-ubuntu:
-    runs-on: ubuntu-20.04
-    if: github.event.ref == 'refs/heads/master' && github.event.repository.full_name == 'vlang/v'
-    timeout-minutes: 30
-    env:
-      V_CI_PERIODIC: 1
-    steps:
-      - uses: actions/checkout@v4
-      - name: Install dependencies 1
-        run: .github/workflows/retry.sh sudo apt-get install --quiet -y sqlite3 libsqlite3-dev
-      - name: Build v
-        run: make
-      - name: Symlink V
-        run: sudo ./v symlink
-        ## - name: Run network tests
-        ##   run: ./v -d network test vlib/net
-
+env:
+  V_CI_PERIODIC: 1
 
-  network-tests-macos:
-    runs-on: macos-14
-    if: github.event.ref == 'refs/heads/master' && github.event.repository.full_name == 'vlang/v'
-    timeout-minutes: 30
-    env:
-      V_CI_PERIODIC: 1
-    steps:
-      - uses: actions/checkout@v4
-      - name: Build V
-        run: make
-      - name: Symlink V
-        run: sudo ./v symlink
-      - name: Ensure thirdparty/cJSON/cJSON.o is compiled, before running tests.
-        run: ./v examples/json.v
-        ## - name: Run network tests
-        ##   run: ./v -d network test vlib/net
-
-  network-windows-msvc:
-    runs-on: windows-2019
-    if: github.event.ref == 'refs/heads/master' && github.event.repository.full_name == 'vlang/v'
-    timeout-minutes: 30
+jobs:
+  network:
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            cc: tcc
+          - os: windows-latest
+            cc: msvc
+          - os: macos-13
+            cc: clang
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    if: github.event_name != 'push' || (github.ref == 'refs/heads/master' && github.repository == 'vlang/v')
     env:
-      V_CI_PERIODIC: 1
-      VFLAGS: -cc msvc
+      VFLAGS: -cc ${{ matrix.cc }} -d network -showcc
```

Comment:
I prefer that `-d network` is passed explicitly only when needed like `v -d network test .`, `v -d network test-self`, and not in VFLAGS.

The reason, is that way it is faster and more intuitive to reproduce the CI failures locally, by just copy/pasting the command from the CI logs, without having to construct it from several different places, or copy several lines.

It is also easier to search where `-d network` is used.

A similar argument applies to `-showcc`. Setting it in VFLAGS, makes it apply to the output of `make` and other unrelated v commands, where it just adds noise 99% of the time.